tests/checks/default-setup-path.fish

```diff
@@ -0,0 +1,12 @@
+#RUN: %fish -C 'set -g fish %fish' %s
+
+# FIXME: Rationalize behavior when PATH is explicitly unset, should this not behave like PATH=""?
```

Comment:
It is very unlikely that that's what people would expect, and a $PATH of "." is basically unusable. If that's *explicitly* set, sure, whatever - but an unset $PATH can also be caused by misconfiguration or e.g. a thing that just clears the environment.

$PATH is also something where we should act like other things because it's a system-wide config knob, and other tools also fall back to their default paths - try `env -u PATH $shell`, which will print some default for bash, dash and zsh.